frontend/src/features/properties/list/buildings/Buildings.tsx

```diff
@@ -12,21 +11,31 @@ export interface IProps {
   data: IProperty[];
   /** Whether to hide the headers. */
   hideHeaders?: boolean;
+  /** Optionally override the columns */
+  columns: ColumnWithProps<IProperty>[];
+  /** handle click event on table building row*/
+  onRowClick?: (data: IProperty) => void;
 }
 
 /**
  * A table displaying a list of buildings.
  * @param {IProps} props Component properties.
  */
-export const Buildings: React.FC<IProps> = ({ data, hideHeaders }) => {
+export const Buildings: React.FC<IProps> = ({
+  data,
+  hideHeaders,
+  columns: externalColumns,
+  onRowClick,
+}) => {
   return (
     <Table<IProperty>
       hideHeaders={hideHeaders}
       name="nestedPropertiesTable"
-      columns={columns}
+      columns={externalColumns}
```

Comment:
Yeah, so having a different set of columns for a table isn't great for % based columns, it makes it very hard to get the columns to align. If this was a html based table it would also be invalid html, but our react tables aren't actually html tables so its fine.

Anyways, updated this to allow the parent to pass in the columns. This allows the columns to be the same for the parent and child in the case of nested building rows.